Dockerfile

```
@@ -0,0 +1,38 @@
+# Stage 1: build frontend
+FROM node:22-alpine AS frontend
+WORKDIR /app
+COPY web/package*.json ./
+RUN npm ci
+COPY web/ ./
+RUN npm run build
+
+# Stage 2: build daemon
+FROM rust:1.88-slim AS builder
+RUN apt-get update && apt-get install -y \
```

Hadolint (2.14.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)